src/univariate/continuous/cosine.jl

```diff
@@ -33,32 +45,34 @@ mode(d::Cosine) = d.μ
 
 var(d::Cosine) = d.σ^2 * 0.13069096604865779  # 0.130... = 1/3 - 2 / π^2
 
-skewness(d::Cosine) = 0.0
+skewness{T<:Real}(d::Cosine{T}) = zero(T)
 
-kurtosis(d::Cosine) = -0.59376287559828102362
+kurtosis{T<:Real}(d::Cosine{T}) = -0.59376287559828102362*one(T)
```

Comment:
Note that we are everywhere promoting integer arguments to Floats when the
parameter is continuous. There are a lot of type instabilities for Ints
lurking in here that we simply couldn't resolve without weird promotion
hijinks everywhere. So given that, Andreas's suggestion is the one to go
with here.

On Thu, Jul 21, 2016 at 4:07 PM Andreas Noack <notifications@github.com>
wrote:

> In src/univariate/continuous/cosine.jl
> <https://github.com/JuliaStats/Distributions.jl/pull/511#discussion_r71777401>
> :
>
> >
> > -kurtosis(d::Cosine) = -0.59376287559828102362
> > +kurtosis{T<:Real}(d::Cosine{T}) = -0.59376287559828102362*one(T)
>
> Wouldn't be correct for BigFloats. On the other hand, my suggestion fails
> for integers. Generic coding can be tricky.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaStats/Distributions.jl/pull/511/files/cb2f87c8124e599d28da1fa8d2af9b4fc3e8d955#r71777401>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADBM5-zNpog2iCUyNZfpBZUzNXw84EYRks5qX9FygaJpZM4JGIxL>
> .
>